cylc/flow/data_store_mgr.py

```diff
@@ -468,7 +469,7 @@ class DataStoreMgr:
     ERR_PREFIX_JOB_NOT_ON_SEQUENCE = 'Invalid cycle point for job: '
 
     def __init__(self, schd, n_edge_distance=1):
-        self.schd = schd
+        self.schd: Scheduler = schd
```

Comment:
```suggestion
        self.schd: 'Scheduler' = schd
```
(However, you might know something I don't, since lacking of quoting here doesn't seem to cause a problem even though Scheduler is only imported when type checking ??)

Reply:
Quotes are needed if the type is not imported [1] or if it is declared further down the file.

> [1] This includes dynamic and `if TYPE_CHECKIKNG` imports.

Reply:
This one is imported in the `if TYPE_CHECKING` block, but weirdly seems to survive non-quoting.

Reply:
Quotes for types imported inside the `TYPE_CHECKING` block are only needed when the type is used in function declarations

Reply:
Ah, that explains it!  (was news to me).